src/v3/src/transformer/terminal/transformEmailMagicLinkOTPOnlyElements.ts

```diff
@@ -85,16 +86,19 @@ export const transformEmailMagicLinkOTPOnly: TerminalKeyTransformer = (transacti
     };
   }
 
-  if (client?.value?.browser && client?.value?.os) {
+  const clientOs = client?.value?.os;
+  const clientBrowser = client?.value?.browser;
+  if (clientBrowser && clientOs) {
+    const isMobileDevice = clientOs === 'Android' || clientOs === 'iOS';
```

Comment:
Check out `src/v3/src/util/browserUtils.ts` -- there are existing utilities to detect both mobile clients so we should use those if possible

Reply:
yeah I think it would make sense if we make an optional transaction parameter where we can pass in a transaction object and check there first

Reply:
instead of the entire transaction, maybe just the context? Or even just the client property from the context. You never know what someone in the future may do with it if the transaction is given to that function 😅

Reply:
haha true good point i think just the client property would be okay

Reply:
Hmm, I'm looking closely at the utility and I'm realizing these _might_ not be compatible, since they are checking different types of inputs (userAgent string vs what comes from `client` object). If you see a good way to handle this gracefully go for it, but I don't see as clear of a need to refactor these into one any more if they are too different.

Reply:
Yeah I think youre right after looking at it more it looks like there isnt really a graceful way to handle it, they check very different inputs.  It could create more confusion than helpfulness with the util function so its probably best not to refactor into one